packages/block-editor/src/components/inserter/index.js

```diff
@@ -370,10 +371,17 @@ export default compose( [
 					blockToInsert = createBlock( allowedBlockType.name );
 				}
 
-				insertBlock( blockToInsert, getInsertionIndex(), rootClientId );
+				insertBlock(
+					blockToInsert,
+					getInsertionIndex(),
+					rootClientId,
+					selectBlockOnInsert
+				);
 
 				if ( onSelectOrClose ) {
-					onSelectOrClose();
+					onSelectOrClose( {
+						insertedBlock: blockToInsert,
```

Comment:
As the block isn't "selected" this is the only reference we have to the block that was just inserted.